### PR TITLE
feat: pull latest changes from upstream

### PR DESCRIPTION
**Pull request by AlexTan331**
feat:
1. pull latest changes from upstream
2. finish lc #53 problem and mark as completed 
3. finish lc #209 problem and mark as completed 
4. finish lc #340 problem and mark as completed

### CODE REVIEW

**Review by MelissaPastore — CHANGES_REQUESTED** [review]
It looks like you edited someone else's files. Please make sure to do your work in your own files. You should be able to copy and paste the code from these files to your own files.

**Comment by AlexTan331** [comment]
> It looks like you edited someone else's files. Please make sure to do your work in your own files. You should be able to copy and paste the code from these files to your own files.

It might sound confusing that I am Taojie Tan while my GitHub name is AlexTan. But I am sure I am editing on my own folder.

File: taojieTan/assignments/slidingwindow/lc209/lc209.py [hunk 1]
```diff
@@ -1,14 +1,53 @@
+# Problem Statement #
+
+# Given an array of positive numbers and a positive number ‘S,’ find the length of the smallest contiguous subarray whose sum is greater than or equal to ‘S’. Return 0 if no such subarray exists.
+
 import math
 
 
 def smallest_subarray_with_given_sum(s, arr):
 
+    #input: positive integer s, positive integer array
```
Comment (MelissaPastore) [inline taojieTan/assignments/slidingwindow/lc209/lc209.py:10]: Nice comments / overview!

File: taojieTan/assignments/slidingwindow/lc340/lc340.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
+# Problem Statement #
```
Comment (MelissaPastore) [inline taojieTan/assignments/slidingwindow/lc340/lc340.py:1]: Again very thorough comments and explanations of your code. Nice work!

**Review by MelissaPastore — APPROVED** [review]

**Pull request merged**